routers/projectRouter.js

```diff
@@ -0,0 +1,98 @@
+//import express
+const express = require("express");
+const router = express.Router();
+
+//importing needed db
+const projectDB = require("../data/helpers/projectModel.js");
+
+//looking for project by id
+//validating a project with that id exists
+router.get("/:id", validateProjectId, (req, res) => {
+    projectDB
+    //searches for specific parameter at specific id
+      .get(req.params.id)
+      .then(project => {
+        res.status(200).json(project);
+      })
+      .catch(error => {
+        console.log("get error", error);
+        res
+          .status(500)
+          .json({
+            error: "There was an error retrieving the project from the database."
+          });
+      });
+  });
+  
+
+//making sure we have data in the body and making sure there is a name and desc.
+//create a post that will have a name and desc.
+router.post("/", validateProjectInfo, (req, res) => {
+  const { name, description } = req.body;
+  projectDB
+    .insert({ name, description })
```

Comment:
Good coding - destructuring the body and then using only those values for the database. If our client sent us info like:
```
{
    name: "Project",
    description: "The next big thing",
    extraprops: "Blargh"
}
```

This would normally cause an error in the server because we have an extra property that we don't need. By destructuring, you're making sure you have only what you need to avoid any strange errors 🔥